tests/v2/validation/charts/gatekeeperallowednamespaces_test.go

```diff
@@ -0,0 +1,177 @@
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, gatekeeperProjectName)
+	n.project = createdProject
+
+	n.gatekeeperChartInstallOptions = &charts.InstallOptions{
+
+		ClusterName: clusterName,
+		ClusterID:   clusterID,
+		Version:     latestGatekeeperVersion,
+		ProjectID:   createdProject.ID,
+	}
+
+}
+func (n *GateKeeperAllowedNamespacesTestSuite) TestGateKeeperAllowedNamespaces() {
+	subSession := n.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := n.client.WithSession(subSession)
+	require.NoError(n.T(), err)
+
+	if !client.Flags.GetValue(environmentflag.GatekeeperAllowedNamespaces) {
+		n.T().Skip()
+	}
+
+	n.T().Log("getting list of all namespaces")
+
+	sysNamespaces := settings.SystemNamespaces.Get()
+	sysNamespacesSlice := strings.Split(sysNamespaces, ",")
+	n.T().Log(sysNamespaces)
+
+	NSKinds := Kinds{
+		{ApiGroups: []string{""},
+			Kinds: []string{"Namespace"}},
+	}
+
+	NSMetadata := Metadata{
+		Name: "ns-must-be-allowed",
+	}
+
+	NSParameters := Parameters{
+		Namespaces: sysNamespacesSlice,
+	}
+
+	NSMatch := Match{
+		Kinds: NSKinds,
+	}
+
+	NSSpec := Spec{
+		EnforcementAction: "deny",
+		Match:             NSMatch,
+		Parameters:        NSParameters,
+	}
+
+	AllowedNamespaces := ConstraintYaml{
+		ApiVersion: "constraints.gatekeeper.sh/v1beta1",
+		Kind:       "K8sAllowedNamespaces",
+		Metadata:   NSMetadata,
+		Spec:       NSSpec,
+	}
+
+	yamlData, err := yaml.Marshal(&AllowedNamespaces)
+	require.NoError(n.T(), err)
+
+	fileName := "allowednamespaces.yaml"
+	err = os.WriteFile(fileName, yamlData, 0644)
+	require.NoError(n.T(), err)
+
+	n.T().Log("Installing latest version of gatekeeper chart")
+	err = charts.InstallRancherGatekeeperChart(client, n.gatekeeperChartInstallOptions)
+	require.NoError(n.T(), err)
+
+	n.T().Log("Waiting for gatekeeper chart deployments to have expected number of available replicas")
+	err = charts.WatchAndWaitDeployments(client, n.project.ClusterID, charts.RancherGatekeeperNamespace, metav1.ListOptions{})
+	require.NoError(n.T(), err)
+
+	n.T().Log("Waiting for gatekeeper chart DaemonSets to have expected number of available nodes")
+	err = charts.WatchAndWaitDaemonSets(client, n.project.ClusterID, charts.RancherGatekeeperNamespace, metav1.ListOptions{})
+	require.NoError(n.T(), err)
+
+	n.T().Log("creating constraint template")
+	readTemplateYamlFile, err := os.ReadFile("./resources/opa-allowednamespacestemplate.yaml")
+	require.NoError(n.T(), err)
+	yamlTemplateInput := &management.ImportClusterYamlInput{
+		DefaultNamespace: charts.RancherGatekeeperNamespace,
+		YAML:             string(readTemplateYamlFile),
+	}
+
+	n.T().Log("creating constraint")
+	readConstraintYamlFile, err := os.ReadFile("./allowednamespaces.yaml")
+	require.NoError(n.T(), err)
+	yamlConstraintInput := &management.ImportClusterYamlInput{
+		DefaultNamespace: charts.RancherGatekeeperNamespace,
+		YAML:             string(readConstraintYamlFile),
```

Comment:
I think you can marshal the struct and then convert it to a string. There is no need to read and write from a file

Reply:
this is now being done with a YAML string instead of creating a file